app/models/concerns/digital_object/persistence.rb

```diff
@@ -42,6 +42,9 @@ def save(manual_commit_to_solr = true)
   def before_save
     # TODO: rewrite with ActiveRecord::Callbacks
     # To be overridden by subclasses
+    # PSEUDO CODE
+    # begin - rescue around the following call, rescue Hyacinth::Exceptions::DataciteError
+    # Question: What to do if above exception is recued/caught?
     mint_and_store_doi(Hyacinth::Datacite::Doi::IDENTIFIER_STATUS[:draft]) if @mint_reserved_doi_before_save || @publish_after_save
```

Comment:
Let's plan to add a `rescue` to this method that will add to the @errors object like this:

```
def before_safe
  ...
  ...
rescue
  @errors.add(:datacite, "This is a helpful error message with information that will be shown to the user.")
end
```